### PR TITLE
output folder only has loc built files

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -105,7 +106,8 @@ async Task BuildOneFile(Document file, Action<Document> buildChild)
 
                 bool ShouldBuildFile(Document file)
                 {
-                    return file.ContentType != ContentType.Unknown && filesBuilder.TryAdd(file);
+                    return file.ContentType != ContentType.Unknown && filesBuilder.TryAdd(file) &&
+                        (docset.FallbackDocset == null/*source docset*/ || file.Docset.FallbackDocset != null /*loc file*/);
```
Comment (live1206) [inline src/docfx/build/Build.cs]: In which case, the file to be built is neither from source docset nor a loc file? And we want to exclude it from build scope

Comment (OsmondJiang) [reply src/docfx/build/Build.cs]: ？

---
In reply to: [227236163](https://github.com/dotnet/docfx/pull/3519#discussion_r227236163) [](ancestors = 227236163)

File: src/docfx/build/Build.cs [hunk 2]
```diff
@@ -69,6 +69,7 @@ Docset GetBuildDocset()
 
         private static async Task<(List<Document> files, DependencyMap sourceDependencies)> BuildFiles(
             Context context,
+            Docset docset,
             HashSet<Document> buildScope,
```
Comment (superyyrrzz) [inline src/docfx/build/Build.cs]: >HashSet<Document> buildScope, [](start = 12, length = 29)

this parameter is no longer needed as it's included in `docset`?

File: src/docfx/build/Build.cs [hunk 3]
```diff
@@ -105,7 +105,10 @@ async Task BuildOneFile(Document file, Action<Document> buildChild)
 
                 bool ShouldBuildFile(Document file)
                 {
-                    return file.ContentType != ContentType.Unknown && filesBuilder.TryAdd(file);
+                    return file.ContentType != ContentType.Unknown &&
+                        filesBuilder.TryAdd(file) &&
+                        !(docset.FallbackDocset != null &&
```
Comment (yufeih) [inline src/docfx/build/Build.cs]: can you refactor this a little bit? there are too much conditions here:
```csharp
// English content in a localized build
if (docset.FallbackDocset != null && file.Docset.LocalizationDocset != null)
{
    return false;
}
return file.ContentType != Unknown && TryAdd(xxx)
```

File: src/docfx/build/toc/BuildTableOfContents.cs [hunk 4]
```diff
@@ -24,11 +24,6 @@ internal static class BuildTableOfContents
             var dependencyMapBuilder = new DependencyMapBuilder();
             var (errors, tocModel, tocMetadata, refArticles, refTocs) = Load(context, file, dependencyMapBuilder);
```
Comment (yufeih) [inline src/docfx/build/toc/BuildTableOfContents.cs]: what if a link in TOC points to an PNG or config? we could have a TODO here.

Comment (OsmondJiang) [reply src/docfx/build/toc/BuildTableOfContents.cs]: toc can't reference to a link or config right now, but i can add a TODO here.

---
In reply to: [227639083](https://github.com/dotnet/docfx/pull/3519#discussion_r227639083) [](ancestors = 227639083)

**Pull request merged**